### DISCUSSION TITLE
How to add multitenancy to Azure Function using Orchard Core Framework

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#10139] [Q&A]
I use out of process Azure Function on .NET 5 and I need to add multitenancy to it.
There is an extension mechanism of middleware similar to ASP.NET Core https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.functions.worker.middleware.ifunctionsworkermiddleware?view=azure-dotnet

How can I reuse Orchard Core Framework for that?

**Comment by jersiovic** [comment]
Maybe in a few months I will be interested on moving some logic of my orchard modules to azure functions. So, the easiest move seems sth like that. Did you managed to do it?
Can you share what you did for making it work?
Thank you

**Comment by ns8482e** [comment]
May be you can create an regular single tenant application as you would normally do for azure functions and have x-tenant-id header in your request.

Based on header you can create tenant scope using orchard framework and run your functions within that scope https://github.com/OrchardCMS/OrchardCore/discussions/9943#discussioncomment-1307909

**Reply by jersiovic** [reply]
Thank you Niraj for the advice, I will take it into account
